sparkly/testing.py

```diff
@@ -135,13 +172,294 @@ def assertDataFrameEqual(self, actual_df, expected_data, fields=None, ordered=Fa
         actual_rows = actual_df.collect() if hasattr(actual_df, 'collect') else actual_df
         actual_data = [row.asDict(recursive=True) for row in actual_rows]
 
-        if ordered:
-            self.assertEqual(actual_data, expected_data)
-        else:
-            try:
-                self.assertCountEqual(actual_data, expected_data)
-            except AttributeError:
-                self.assertItemsEqual(actual_data, expected_data)
+        return self.assertEqual(
+            actual_data,
+            expected_data,
+            ignore_order=not ordered,
+            ignore_order_depth=1,
+            rtol=0,
+            equal_nan=False,
+        )
+
+    def assertEqual(self,
```

Comment:
> this would introduce a breaking change in the API since we have already defined `assertDataFrameEqual`

Why is this a breaking change?

> if you want to be able to compare lists of floats right? So I don't understand what you're trying to say with this comparison.

Why `sparkly` should take care about such comparators? The library is designed to work with `pyspark`. The main data structure is `DataFrame`. There are different `assert` functions to work with different data structures, we can't (and shouldn't) cover all the world.